game-core/src/main/java/games/strategy/engine/framework/startup/login/HmacSha512Authenticator.java

```diff
@@ -41,31 +41,36 @@ private HmacSha512Authenticator() {}
    *         node; never {@code null}.
    */
   static Map<String, String> newChallenge() {
+    // This length is relatively large for a nonce simply because we do not currently expire nonces after they have been
+    // issued. If this were done, and a short nonce lifetime was used, we could probably get away with 8 bytes of random
+    // data safely as long as we included the timestamp as part of the nonce.
+    //
+    // https://security.stackexchange.com/questions/1952/how-long-should-a-random-nonce-be
+    final int nonceLengthInBytes = 64;
+
+    // Salt length should match hash output size.
```

Comment:
I see value in this comment to indicate that the values are set for different reasons. 'Hash output size' raises a lot more questions than it answers, a problem of hidden context. It is not clear exactly which hash output is being referenced from the TripleA code base. Hence why it was struck.